examples/triage/baseaddress.cpp

```diff
@@ -0,0 +1,427 @@
+		emit ResultReady(results);
+		return;
+	}
+
+	if (lowerBoundary >= upperBoundary)
+	{
+		results.Status = "Upper boundary address is less than lower";
+		emit ResultReady(results);
+		return;
+	}
+
+	if (!BinaryNinja::BinaryView::ParseExpression(
+		m_view, m_inputs->MaxPointersPerCluster->text().toStdString(), value, 0, errorStr))
+	{
+		results.Status = "Invalid max pointers (" + errorStr + ")";
+		emit ResultReady(results);
+		return;
+	}
+
+	uint32_t maxPointersPerCluster = value;
+	if (maxPointersPerCluster < 2)
+	{
+		results.Status = "Invalid max pointers (must be >= 2)";
+		emit ResultReady(results);
+		return;
+	}
+
+	BNBaseAddressDetectionPOISetting poiSetting = BaseAddressDetectionPOISettingFromString(
+		m_inputs->POIBox->currentText().toStdString());
+	BinaryNinja::BaseAddressDetectionSettings settings = {
+		m_inputs->ArchitectureBox->currentText().toStdString(),
+		m_inputs->AnalysisBox->currentText().toStdString(),
+		minStrlen,
+		alignment,
+		lowerBoundary,
+		upperBoundary,
+		poiSetting,
+		maxPointersPerCluster,
+	};
+
+	if (!m_baseDetection->DetectBaseAddress(settings))
+		emit ResultReady(results);
+
+	auto scores = m_baseDetection->GetScores(&results.Confidence, &results.LastTestedBaseAddress);
+	results.Scores = scores;
+	for (const auto& score : scores)
+	{
+		auto reasons = m_baseDetection->GetReasonsForBaseAddress(score.second);
+		results.Reasons[score.second] = reasons;
+	}
+
+	emit ResultReady(results);
+}
+
+
+void BaseAddressDetectionWidget::HideResultsWidgets(bool hide)
+{
+	if (hide)
+	{
+		m_preferredBaseLabel->setHidden(true);
+		m_preferredBase->setHidden(true);
+		m_confidenceLabel->setHidden(true);
+		m_confidence->setHidden(true);
+		m_resultsTableWidget->setHidden(true);
+		m_reloadBase->setHidden(true);
+		m_rebaseButton->setHidden(true);
+	}
+	else
+	{
+		m_preferredBaseLabel->setHidden(false);
+		m_preferredBase->setHidden(false);
+		m_confidenceLabel->setHidden(false);
+		m_confidence->setHidden(false);
+		m_resultsTableWidget->setHidden(false);
+		m_reloadBase->setHidden(false);
+		m_rebaseButton->setHidden(false);
+	}
+}
+
+
+void BaseAddressDetectionWidget::HandleResults(const BaseAddressDetectionQtResults& results)
+{
+	if (!results.Status.empty())
+		m_status->setText(QString::fromStdString(results.Status));
+
+	if (results.Status.empty() && m_worker->IsAborted())
+		m_status->setText(QString("Aborted by user (Last Base: 0x%1)").arg(
+			QString::number(results.LastTestedBaseAddress, 16)));
+
+	if (results.Scores.empty())
+	{
+		if (!m_worker->IsAborted() && results.Status.empty())
+			m_status->setText("Completed with no results");
+		m_preferredBase->setText("Not available");
+		m_confidence->setText("Not available");
+	}
+	else
+	{
+		HideResultsWidgets(false);
+		if (results.Status.empty() && !m_worker->IsAborted())
+			m_status->setText("Completed with results");
+		m_preferredBase->setText(QString("0x%1").arg(QString::number(results.Scores.rbegin()->second, 16)));
```

Comment:
nit: No need to convert to string `.arg` can handle that for you `QString("0x%1").arg(results.Scores.rbegin()->second, 0, 16)`

Reply:
The `QString::number` call is needed. It's converting the preferred base address to a hex string. If I change it to your recommendation I get the following:

![image](https://github.com/Vector35/binaryninja-api/assets/14361933/939a2f39-f129-4e2b-a162-5faa81fbe855)

Reply:
Ah, that did it. I mis-read 😅